lib/miner-info.js

```diff
@@ -28,10 +35,39 @@ async function getChainHead ({ maxAttempts = 5 } = {}) {
  * @param {number} [options.maxAttempts]
  * @returns {Promise<string>} Miner's PeerId, e.g. `12D3KooWMsPmAA65yHAHgbxgh7CPkEctJHZMeM3rAvoW8CZKxtpG`
  */
-export async function getMinerPeerId (minerId, { maxAttempts = 5 } = {}) {
-  const chainHead = await getChainHead({ maxAttempts })
+export async function getMinerPeerId (minerId, { maxAttempts = 5, smartContract, rpcFn } = {}) {
+  // Make a concurrent request to both sources: FilecoinMinerInfo and smart contract
+  const [minerInfoResult, contractResult] = await Promise.allSettled([
+    getMinerPeerIdFromFilecoinMinerInfo(minerId, { maxAttempts, rpcFn }),
+    getMinerPeerIdFromSmartContract(minerId, { smartContract })
+  ])
+  // Check contract result first
+  if (contractResult.status === 'fulfilled' && contractResult.value && contractResult.value !== '') {
```

Comment:
FWIW, `contractResult.value` is enough of a check.

```suggestion
  if (contractResult.status === 'fulfilled' && contractResult.value) {
```

JavaScript considers an empty string as a false value.

```
❯ node
Welcome to Node.js v22.13.0.
Type ".help" for more information.
> const val = ''
undefined
> if (val) console.log('not empty')
undefined
> const val2 = 'val'
undefined
> if (val2) console.log('not empty')
not empty
```